art/attacks/evasion/pixel_threshold.py

```diff
@@ -0,0 +1,1458 @@
+        # np.inf). If it has you have to calculate the initial energies
+        if np.all(np.isinf(self.population_energies)):
+            self._calculate_population_energies()
+
+        if self.dither is not None:
+            self.scale = (self.random_number_generator.rand()
+                          * (self.dither[1] - self.dither[0]) + self.dither[0])
+
+        ##############
+        # CHANGES: self.func operates on the entire parameters array
+        ##############
+
+        itersize = max(0, min(self.num_population_members,
+                              self.maxfun - self._nfev + 1))
+        trials = np.array([self._mutate(c)
+                           for c in range(itersize)])  # TODO:can be vectorized
+        for trial in trials:
+            self._ensure_constraint(trial)
+        parameters = np.array([self._scale_parameters(trial)
+                               for trial in trials])
+        energies = self.func(parameters, *self.args)
+        self._nfev += itersize
+
+        for candidate, (energy, trial) in enumerate(zip(energies, trials)):
+            # if the energy of the trial candidate is lower than the
+            # original population member then replace it
+            if energy < self.population_energies[candidate]:
+                self.population[candidate] = trial
+                self.population_energies[candidate] = energy
+
+                # if the trial candidate also has a lower energy than the
+                # best solution then replace that as well
+                if energy < self.population_energies[0]:
+                    self.population_energies[0] = energy
+                    self.population[0] = trial
+
+        # for candidate in range(self.num_population_members):
+        #     if self._nfev > self.maxfun:
+        #         raise StopIteration
+
+        #     # create a trial solution
+        #     trial = self._mutate(candidate)
+
+        #     # ensuring that it's in the range [0, 1)
+        #     self._ensure_constraint(trial)
+
+        #     # scale from [0, 1) to the actual parameter value
+        #     parameters = self._scale_parameters(trial)
+
+        #     # determine the energy of the objective function
+        #     energy = self.func(parameters, *self.args)
+        #     self._nfev += 1
+
+        #     # if the energy of the trial candidate is lower than the
+        #     # original population member then replace it
+        #     if energy < self.population_energies[candidate]:
+        #         self.population[candidate] = trial
+        #         self.population_energies[candidate] = energy
+
+        #         # if the trial candidate also has a lower energy than the
+        #         # best solution then replace that as well
+        #         if energy < self.population_energies[0]:
+        #             self.population_energies[0] = energy
+        #             self.population[0] = trial
+
+        ##############
+        ##############
```

Comment:
Could those comments be removed?

Reply:
Hi @mathsinn, I am not sure whether removing these lines can be removed. All the lines in the file from `line number 454` is modified from SciPy's implementation of Differential Evolution. 
As you can see in the original implementation, these lines are used; however, in the modification, these lines are commented out. 

https://github.com/scipy/scipy/blob/70e61dee181de23fdd8d893eaa9491100e2218d7/scipy/optimize/_differentialevolution.py#L695

Therefore this comment comes under the changes done to the original code. 

I would prefer if you can keep this commented section, so people have a clear indication of how and what changes are done to the original code.